### PR TITLE
[CodePush] Migration of interfaces

### PR DESCRIPTION
**Pull request by ruslan-bikkinin**

### CODE REVIEW

File: sdk/appcenter-codepush/src/test/java/com/microsoft/appcenter/codepush/DownloadProgressTest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+
+public class DownloadProgressTest {
+
+    private DownloadProgress downloadProgress;
+
+    @Before
+    public void setUp(){
```
Comment (MatkovIvan) [inline sdk/appcenter-codepush/src/test/java/com/microsoft/appcenter/codepush/DownloadProgressTest.java]: Missing space

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/DownloadProgress.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ */
+public class DownloadProgress {
+    private long mTotalBytes;
+    private long mReceivedBytes;
+
+    /**
+     * DownloadProgress constructor.
+     *
+     * @param totalBytes long Total size of update in bytes.
+     * @param receivedBytes long Amount of bytes already downloaded from server.
```
Comment (MatkovIvan) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/DownloadProgress.java]: remove `long`

File: sdk/appcenter-codepush/src/test/java/com/microsoft/appcenter/codepush/DownloadProgressTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        downloadProgress = new DownloadProgress(1024, 512);
+    }
+
+    @Test
+    public void testGetTotalBytes(){
```
Comment (sergey-akhalkov) [inline sdk/appcenter-codepush/src/test/java/com/microsoft/appcenter/codepush/DownloadProgressTest.java]: missing space `(){`

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/DownloadProgress.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+     */
+    public DownloadProgress(long totalBytes, long receivedBytes){
+        mTotalBytes = totalBytes;
+        mReceivedBytes = receivedBytes;
+    }
+
+    /**
+     * Returns total size of update in bytes.
+     *
+     * @return total size of update in bytes.
```
Comment (MatkovIvan) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/DownloadProgress.java]: Start with capital letter

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/interfaces/CodePushBinaryVersionMismatchListener.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package com.microsoft.appcenter.codepush.interfaces;
+
+import com.microsoft.appcenter.codepush.datacontracts.CodePushRemotePackage;
+
+/**
+ * Interface for listener of binary version mismatch event.
+ */
+public interface CodePushBinaryVersionMismatchListener {
+    void binaryVersionMismatchChanged(CodePushRemotePackage update);
```
Comment (sergey-akhalkov) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/interfaces/CodePushBinaryVersionMismatchListener.java]: no comments for `binaryVersionMismatchChanged` method

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/DownloadProgress.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
+package com.microsoft.appcenter.codepush;
+
+/**
+ * Object representing progress of update download.
+ */
+public class DownloadProgress {
+    private long mTotalBytes;
+    private long mReceivedBytes;
```
Comment (sergey-akhalkov) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/DownloadProgress.java]: no comments for `private` props

Comment (MatkovIvan) [reply sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/DownloadProgress.java]: Empty line between members

Comment (annakocheshkova) [reply sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/DownloadProgress.java]: I have added this class in my pull, see [task](https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems?id=31373&_a=edit)

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/DownloadProgress.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+     *
+     * @return Amount of bytes already downloaded from server.
+     */
+    public long getReceivedBytes() {
+        return mReceivedBytes;
+    }
+
+    /**
+     * Indicates whether downloading of update is completed.
+     * @return True if downloading of update is completed, false otherwise.
```
Comment (MatkovIvan) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/DownloadProgress.java]: Missing empty line before

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/interfaces/CodePushBinaryVersionMismatchListener.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+import com.microsoft.appcenter.codepush.datacontracts.CodePushRemotePackage;
+
+/**
+ * Interface for listener of binary version mismatch event.
+ */
+public interface CodePushBinaryVersionMismatchListener {
+
+    /**
+     * Callback for handling binary version mismatch event.
+     * @param update Remote package (from server).
```
Comment (MatkovIvan) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/interfaces/CodePushBinaryVersionMismatchListener.java]: Missing empty line before

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/interfaces/CodePushDownloadProgressListener.java [hunk 9]
```diff
@@ -0,0 +1,14 @@
+package com.microsoft.appcenter.codepush.interfaces;
+
+/**
+ * Interface for listener of progress of update event.
+ */
+public interface CodePushDownloadProgressListener {
+
+    /**
+     * Callback for handling download progress changed event.
+     * @param receivedBytes Total size of update in bytes.
```
Comment (MatkovIvan) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/interfaces/CodePushDownloadProgressListener.java]: Missing empty line before

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/interfaces/CodePushSyncStatusListener.java [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+import com.microsoft.appcenter.codepush.enums.CodePushSyncStatus;
+
+/**
+ * Interface for listener of sync status event.
+ */
+public interface CodePushSyncStatusListener {
+
+    /**
+     * Callback for handling sync status changed event.
+     * @param syncStatus Status of sync.
```
Comment (MatkovIvan) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/interfaces/CodePushSyncStatusListener.java]: Missing empty line before

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/interfaces/DownloadProgressCallback.java [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+import com.microsoft.appcenter.codepush.DownloadProgress;
+
+/**
+ * Interface for download progress of update callback.
+ */
+public interface DownloadProgressCallback {
+
+    /**
+     * Callback function for handling progress of downloading of update.
+     * @param downloadProgress Progress of downloading of update.
```
Comment (MatkovIvan) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/interfaces/DownloadProgressCallback.java]: Missing empty line before

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/DownloadProgress.java (hunk 6 above)
Comment (ruslan-bikkinin) [reply sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/DownloadProgress.java]: @annakocheshkova unfortunately I couldn't migrate interfaces without this one, so I had to include it in my PR + test for it. Please, sync your branch with mine to resolve this issue.

**Pull request merged**